iohub/reader.py

```diff
@@ -154,30 +193,127 @@ def imread(
     logging.getLogger(__name__)
 
     # try to guess data type
-    if data_type is None:
-        if ngff_version := _check_zarr_data_type(path):
-            data_type = "omezarr"
-        elif _check_ndtiff(path):
-            data_type = "ndtiff"
-        elif _check_multipage_tiff(path):
-            data_type = "ometiff"
-        elif _check_single_page_tiff(path):
-            data_type = "singlepagetiff"
-        else:
-            raise FileNotFoundError(
-                "Failed to infer data type: "
-                f"No compatible data found under {path}."
-            )
+    extra_info = None
+    if not data_type:
+        data_type, extra_info = _infer_format(path)
     # identify data structure type
     if data_type == "ometiff":
         return MicromanagerOmeTiffReader(path, extract_data)
     elif data_type == "singlepagetiff":
         return MicromanagerSequenceReader(path, extract_data)
     elif data_type == "omezarr":
-        return ZarrReader(path, version=ngff_version)
+        if extra_info is None:
+            _, extra_info = _infer_format(path)
+        if extra_info == "0.4":
+            # `warnings` instead of `logging` since this can be avoided
+            warnings.warn(
+                UserWarning(
+                    "For NGFF v0.4 datasets, `iohub.open_ome_zarr()` "
+                    "is preferred over `iohub.imread()`. "
+                    "`iohub.imread()` will return an NGFFNode object "
```

Comment:
imread -> read_micromanager here in these two lines

Reply:
Thanks for the catch, fixing.

Reply:
Actually this warning message is more outdated than just the name...